src/cjson/lua_cjson.c

```diff
@@ -1392,12 +1407,43 @@ static int json_decode(lua_State *l)
 {
     json_parse_t json;
     json_token_t token;
+    json_options_t options = { .object_use_luanil = false, .array_use_luanil = false };
+
+
     size_t json_len;
 
-    luaL_argcheck(l, lua_gettop(l) == 1, 1, "expected 1 argument");
+    switch (lua_gettop(l)) {
+    case 1:
+        break;;
+    case 2:
```

Comment:
@justinmk not sure if there is a better way to do this

Reply:
?

Reply:
I wasn't sure if there was a cleaner way to check the string membership of the table in lua. In lua you can just do if table["string"], wasn't sure if I could use something like get field and check if it was not nil.

Reply:
That example doesn't make sense actually, the signature of lua_gettable is:
```
LUA_API void  (lua_gettable) (lua_State *L, int idx);
```
It just pushes the entry to the top of the stack.